lnwallet/rpcwallet/remote_signer.go

```diff
@@ -0,0 +1,93 @@
+package rpcwallet
+
+import (
+	"context"
+	"time"
+
+	"github.com/lightningnetwork/lnd/lnrpc/signrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"google.golang.org/grpc"
+)
+
+type (
+	StreamClient = walletrpc.WalletKit_SignCoordinatorStreamsClient
+	StreamServer = walletrpc.WalletKit_SignCoordinatorStreamsServer
+)
+
+// RemoteSigner is an interface that abstracts the communication with a remote
+// signer. It extends the RemoteSignerRequests, and adds some additional methods
+// to manage the connection and verify the health of the remote signer.
+type RemoteSigner interface {
+	// RemoteSignerRequests is an interface that defines the requests that
+	// can be sent to a remote signer.
+	RemoteSignerRequests
+
+	// Timeout returns the set connection timeout for the remote signer.
+	Timeout() time.Duration
+
+	// Ready blocks and returns nil when the remote signer is ready to
+	// accept requests.
+	Ready() error
+
+	// Ping verifies that the remote signer is still responsive.
+	Ping(timeout time.Duration) error
+
+	// Run feeds lnd with the incoming stream set up by an outbound remote
+	// signer and then blocks until the stream is closed. Lnd can then send
+	// any requests to the remote signer through the stream.
+	Run(stream StreamServer) error
```

Comment:
This is used in `lnrpc/walletrpc/walletkit_server.go`, so it needs to exist. I did add a separate interface `InboundRemoteSignerConnection` that adds this functionality, which is only implemented by the instance that handles connections from outbound remote signers on the watch-only node. So that should be addressed as well now :)